lms/djangoapps/bulk_email/tasks.py

```diff
@@ -387,25 +393,40 @@ def _filter_optouts_from_recipients(to_list, course_id):
     return to_list, num_optout
 
 
-def _get_source_address(course_id, course_title):
+def _get_source_address(user_pk, course_id, course_title):
     """
     Calculates an email address to be used as the 'from-address' for sent emails.
 
-    Makes a unique from name and address for each course, e.g.
+    Makes a unique from name and address for each course,
 
         "COURSE_TITLE" Course Staff <coursenum-no-reply@courseupdates.edx.org>
 
+    The string is localized based on the user's language preference.
     """
     course_title_no_quotes = re.sub(r'"', '', course_title)
 
     # For the email address, get the course.  Then make sure that it can be used
     # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
     # character appears.
-    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
-        course_title_no_quotes,
-        re.sub(r"[^\w.-]", '_', course_id.course),
-        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
+
+    try:
+        user = User.objects.get(pk=user_pk)
```

Comment:
I see what the problem is. However it is stored in yet another table:

    mysql> select * from user_api_userpreference;
    +----+---------+-----------+-------+
    | id | user_id | key       | value |
    +----+---------+-----------+-------+
    |  1 |       1 | pref-lang | ar    |
    |  2 |       3 | pref-lang | ar    |
    |  5 |       2 | dark-lang | en    |
    |  6 |       5 | pref-lang | ar    |
    |  7 |       6 | pref-lang | ar    |
    +----+---------+-----------+-------+

Not sure if it is good or even possible to perform joins to get this at the database level instead of using the function `get_user_preference`